lib/oauth2/client.rb

```diff
@@ -136,8 +136,11 @@ def get_token(params, access_token_opts = {}, access_token_class = AccessToken)
         opts[:params] = params
       end
       response = request(options[:token_method], token_url, opts)
-      error = Error.new(response)
-      raise(error) if options[:raise_errors] && !(response.parsed.is_a?(Hash) && response.parsed['access_token'])
+      if options[:raise_errors] && !(response.parsed.is_a?(Hash) &&
```

Comment:
This line is a little tough to read. I'd change to something like this:

```ruby
response_contains_token = response.parsed.is_a?(Hash) &&
  (response.parsed['access_token'] || response.parsed['id_token'])

error = Error.new(response)
raise(error) if options[:raise_errors] && !(response_contains_token)
```

Also this method now breaks the lint check, which is failing the build. In a method above there's a comment to disable a few extra Rubocops, I'd add the same comment to this method `def` as well:

```ruby
# rubocop:disable CyclomaticComplexity, MethodLength, Metrics/AbcSize
```